src/hooks/transactions/useCheckTransactionStatus/useCheckTransactionStatus.ts

```diff
@@ -0,0 +1,197 @@
+        errorMessage: resultWithError?.returnMessage
+      })
+    );
+    dispatch(
+      updateSignedTransactions({
+        sessionId,
+        status: TransactionBatchStatusesEnum.fail,
+        errorMessage: resultWithError?.returnMessage
+      })
+    );
+  };
+}
+
+function useManageFailedTransactions() {
+  const dispatch = useDispatch();
+  return (sessionId: string) => {
+    dispatch(
+      updateSignedTransactions({
+        sessionId,
+        status: TransactionBatchStatusesEnum.timedOut
+      })
+    );
+  };
+}
+
+export function useCheckTransactionStatus() {
+  const dispatch = useDispatch();
+  const retriesRef = useRef<RetriesType>({});
+  const timeoutRefs = useRef<string[]>([]);
+  const manageFailedTransactions = useManageFailed();
+  const manageTimedOutTransactions = useManageFailedTransactions();
+  const { pendingTransactionsArray } = useGetPendingTransactions();
+
+  function checkTransactionStatus(
+    getTransactionsByHash?: GetTransactionsByHashesType
+  ) {
+    const pendingBatches = pendingTransactionsArray.filter(
+      ([sessionId, transactionBatch]) => {
+        const isPending =
+          sessionId != null && getIsTransactionPending(transactionBatch.status);
+        return isPending;
+      }
+    );
+    if (pendingBatches.length > 0) {
+      pendingBatches.forEach(([sessionId, transactionBatch]) => {
+        checkBatch({
+          sessionId,
+          transactionBatch,
+          getTransactionsByHash
+        });
+      });
+    }
+  }
+
+  async function checkBatch({
+    sessionId,
+    transactionBatch: { transactions, status, customTransactionInformation },
+    getTransactionsByHash = defaultGetTxByHash
+  }: TransactionStatusTrackerPropsType) {
+    const isPending = sessionId != null && getIsTransactionPending(status);
+    try {
+      if (!isPending || transactions == null) {
+        return;
+      }
+
+      const pendingTransactions = getPendingTransactions(
+        transactions,
+        timeoutRefs.current
+      );
+
+      if (pendingTransactions?.length === 0) {
+        return;
+      }
+      const serverTransactions = await getTransactionsByHash(
+        pendingTransactions
+      );
+      for (const serverTransaction of serverTransactions) {
+        const {
+          hash,
+          status,
+          results,
+          invalidTransaction,
+          pendingResults,
+          hasStatusChanged
+        } = serverTransaction;
+        try {
+          const retriesForThisHash = retriesRef.current[hash];
+          if (retriesForThisHash > 30) {
+            // consider transaction as stuck after 1 minute
+            manageTimedOutTransactions(sessionId);
+            return;
+          }
+
+          if (invalidTransaction || getIsTransactionPending(status)) {
+            retriesRef.current[hash] = retriesRef.current[hash]
+              ? retriesRef.current[hash] + 1
+              : 1;
+            return;
+          }
+
+          if (!pendingResults) {
```

Comment:
pending results doesn't exist anymore, a transaction will enter success state only after the sc call executes, we can drop this